core/ledger/kvledger/channelinfo_provider.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package kvledger
+
+import (
+	cb "github.com/hyperledger/fabric-protos-go/common"
+	"github.com/hyperledger/fabric/bccsp/factory"
+	"github.com/hyperledger/fabric/common/channelconfig"
+	"github.com/hyperledger/fabric/common/ledger/blkstorage"
+	"github.com/hyperledger/fabric/protoutil"
+)
+
+// ChannelInfoProvider implements ledger.ChannelInfoProvider interface
+type ChannelInfoProvider struct {
+	BlkStoreProvider *blkstorage.BlockStoreProvider
+}
+
+// GetAllMSPIDs retrieves the MSPIDs of application organizations in all the channel configurations,
+// including current and previous channel configurations.
+func (p *ChannelInfoProvider) GetAllMSPIDs(channelName string) ([]string, error) {
+	blockStore, err := p.BlkStoreProvider.Open(channelName)
+	if err != nil {
+		return nil, err
+	}
+	blockchainInfo, err := blockStore.GetBlockchainInfo()
+	if err != nil {
+		return nil, err
+	}
+
+	// Iterate over the config blocks to get all the channel configs, extract MSPIDs and add to mspidsMap
+	mspidsMap := map[string]string{}
+	blockNum := blockchainInfo.Height
+	for blockNum > 0 {
+		configBlock, configBlockNum, err := mostRecentConfigBlockBelow(blockNum, blockStore)
+		if err != nil {
+			return nil, err
+		}
+		channelConfig, err := channelconfig.ExtractChannelConfig(configBlock, factory.GetDefault())
+		if err != nil {
+			return nil, err
+		}
+
+		orgs := channelConfig.ApplicationConfig().Organizations()
+		for orgName, appOrg := range orgs {
+			mspid := appOrg.MSPID()
+			if _, ok := mspidsMap[mspid]; !ok {
+				mspidsMap[mspid] = orgName
+			}
+		}
+		blockNum = configBlockNum
+	}
+
+	mspids := make([]string, 0, len(mspidsMap))
```

Comment:
0 is needed to set the initial size because the implementation is using "append". If 0 is removed, we need to add an index to set each element because initial size becomes len(mspidsMap).

Reply:
I see. Got it. IMO, we can just go with a simple `var mspids []string`